### ISSUE TITLE
torch.nn.functional.silu loses a representable float64 second-derivative signal

### ISSUE THREAD

**Issue by ALinrunrun** [pytorch/pytorch#195682]
### 🐛 Describe the bug

`torch.nn.functional.silu` returns `0.0` for a second-order derivative whose mathematically correct value is finite, normal, and nonzero in `float64`.

SiLU is defined as `x * sigmoid(x)`. At `x = 40.410841513445845`, the high-precision reference second derivative rounds to approximately `-1.0820525268131133e-16`, which is finite and representable in `float64`. However, PyTorch returns `0.0`.

This issue reproduces on CPU, so CUDA availability does not affect the result.

```python
import torch
import torch.nn.functional as F

dtype = torch.float64

x = torch.tensor(40.410841513445845, dtype=dtype, requires_grad=True)

y = F.silu(x)
g1, = torch.autograd.grad(y, x, create_graph=True)
actual, = torch.autograd.grad(g1, x)

expected = torch.tensor(-1.0820525268131133e-16, dtype=dtype)

print("actual:")
print(actual)
print("expected:")
print(expected)
```

### Actual vs expected result

```text
actual:
tensor(0., dtype=torch.float64)

expected:
tensor(-1.0821e-16, dtype=torch.float64)
```

The discrepancy occurs in the second-order backward pass. The expected second derivative is a finite, normal, nonzero `float64` value, but PyTorch collapses it to `0.0`.

### Versions

```text
PyTorch version: 2.12.0+cu130
Is debug build: False
CUDA used to build PyTorch: 13.0
ROCM used to build PyTorch: N/A

OS: Ubuntu 24.04.3 LTS (x86_64)
Python version: 3.13.13 | packaged by Anaconda, Inc.
Python platform: Linux x86_64

Is CUDA available: False
GPU models:
  GPU 0: NVIDIA RTX 6000 Ada Generation
  GPU 1: NVIDIA RTX 6000 Ada Generation
Nvidia driver version: 570.211.01

CPU:
  Model name: AMD Ryzen Threadripper PRO 7985WX 64-Cores
  CPU(s): 128

Versions of relevant libraries:
[pip3] numpy==2.4.6
[pip3] torch==2.12.0
[pip3] triton==3.7.0
```

cc @ezyang @albanD @gqchen @nikitaved @soulitzer @Varal7 @bobrenjc93 @mruberry @jbschlosser @walterddr @mikaylagawarecki
